ocpmodels/models/forcenet_sequential.py

```diff
@@ -609,27 +620,22 @@ def __init__(
         )
         self.seq_model = nn.Sequential(*list_of_modules)
 
-    def forward(self, data):
-
-        atomic_numbers = data.atomic_numbers
-        pos = data.pos
-        batch = data.batch
-        edge_index = data.edge_index
-        cell = data.cell
-        cell_offsets = data.cell_offsets
-        neighbors = data.neighbors
+        # number of sequential modules and sum of balance params
+        # should be the same!
+        assert len(self.seq_model) == sum(pipeline_parallel_balance)
 
-        input = (
-            atomic_numbers,
-            pos,
-            batch,
-            edge_index,
-            cell,
-            cell_offsets,
-            neighbors,
+        self.pipe_model = GPipe(
+            self.seq_model,
+            balance=pipeline_parallel_balance,
+            devices=pipeline_parallel_devices,
+            chunks=pipeline_parallel_chunks,
+            checkpoint=pipeline_parallel_checkpoint,
```

Comment:
With Gpipe, it seems that checkpointing is either on or off, maybe you have some control with the number of chunks. It doesn't seem like there is an easy way to checkpoint only the edge mlp, for instance. Is that right?

Reply:
Yes, checkpointing with gpipe is at the level of individual partitions. 
And you're right that there isn't an easy way to checkpoint just the edge mlp.